gaia-fast-list.js

```diff
@@ -772,6 +785,38 @@ Internal.prototype = {
       : '0';
   },
 
+  populateItemDetail(el, i) {
+    if (!this.getItemImageSrc) return;
+
+    var img = this.getItemImage(el);
+    if (!img) return;
+
+    var record = this.getRecordAt(i);
+    Promise.resolve(this.getItemImageSrc(record, i))
+      .then(src => {
+
+        // There is a chance that the item
+        // could have been recycled before
+        // the user was able to fetch the image.
+        // Abort here if that's the case.
+        if (el.dataset.index != i) return debug('item recycled');
```

Comment:
I don't have any issue with being extra safe here.
But I don't think it can happen. Since this should be guaranteed to run before the next task (a scroll event triggering some recycling).